wdl/wdlParser.py

```diff
@@ -231,6 +232,53 @@ def module():
 
     return mod_type
 
+
+# -----------------------------------------------------------------------------
+# @fn     lvds
+# @brief  rules for the LVDSL keyword
+# @param  string slotNumber
+# @param  int type
+# @return none, appends to global variable "lvdsOutput"
+# -----------------------------------------------------------------------------
+def lvds(slot_number):
+    """
+    These are the rules for the LVDSL keyword, encountered while parsing
+    the SLOT command for the modules (.mod) file. Required format is
+    LVDSL # "label";
+
+    where # is any number
+    where "label" is a string
+    """
+    global token
+    global lvdsOutput
+
+    lvds_chan = None
+
+    if found(NUMBER):
+        lvds_chan = token.cargo
+        if module_name.upper() in {"LVDS"}:
+            if not (1 <= int(lvds_chan) <= 16):
+                error(
+                    f"LVDS channel {dq(lvds_chan)} outside range {{1:16}} for module: "
+                    f"{module_name.upper()}"
+                )
+        else:
+            error(f"LVDSL is an invalid keyword for module: {module_name.upper()}")
+
+    consume(NUMBER)
+
+    # there can be an optional label, specified as token type=STRING
+    if found(STRING):
+        label = token.cargo[1:-1]  # strip leading and trailing quote chars
+        consume(STRING)
+    else:
+        label = ""
+    consume(";")
+
+    if label != "":
+        lvdsOutput += "MOD" + slot_number + "\LVDS_LABEL" + lvds_chan + "=" + label + "\n"
```

Comment:
I'm sure you know this, and it doesn't matter at all, but :
`lvdsOutput+= f"MOD{slot_number}\LVDS_LABEL{lvds_chan}={label}\n" ` 
executes about 100x faster than this line since strings are immutable in python so you have 7 string copies here at the moment

Reply:
I agree with all these suggestions, but at this point I'm trying to be the tailor that cuts little.  I think we can address all these when we do a refactor.  Good to have them documented here.